lib/src/parse/parser.dart

```diff
@@ -517,24 +517,39 @@ class Parser {
     return true;
   }
 
-  /// Consumes the next character if it's equal to [letter], ignoring ASCII
-  /// case.
+  /// Consumes the next character or escape sequence if it matches [expected].
+  ///
+  /// Matching will be case-insensitive unless [caseSensitive] is true.
   @protected
-  bool scanCharIgnoreCase(int letter) {
-    if (!equalsLetterIgnoreCase(letter, scanner.peekChar())) return false;
-    scanner.readChar();
-    return true;
+  bool scanIdentChar(int char, {bool caseSensitive = false}) {
+    bool matches(int actual) {
+      if (caseSensitive) return actual == char;
+      if (char >= $a && char <= $z) return equalsLetterIgnoreCase(char, actual);
+      return characterEqualsIgnoreCase(char, actual);
```

Comment:
I forgot we had this function! Why not just use this instead of toggling between it and `equalsLetterIgnoreCase()`? It's already optimized to work with either letters or not.